napari/utils/perf/_timers.py

```diff
@@ -52,24 +56,35 @@ def __init__(self):
         self.trace_file = None
 
     def add_event(self, event: PerfEvent):
-        """Add one timing event.
+        """Add one completed event.
 
         Parameters
         ----------
         event : PerfEvent
             Add this event.
         """
-        # Write if actively tracing.
+        # Add event if tracing.
         if self.trace_file is not None:
-            self.trace_file.write_event(event)
+            self.trace_file.add_event(event)
 
-        # Update our self.timers (in milliseconds).
-        name = event.name
-        duration_ms = event.duration_ms
-        if name in self.timers:
-            self.timers[name].add(duration_ms)
-        else:
-            self.timers[name] = Stat(duration_ms)
+        if event.type == "X":
```

Comment:
ie here you would write `if type(event) == PerfEvent`

Reply:
Addressed above.